### PR TITLE
10 implement authentification

### LINKED ISSUE #31: US1-T7 - Implement the Authentication
Related to #19  (US2)

### PR DESCRIPTION
**Pull request by LouisChoiniere**
Implement the Authentication of student and teacher
Modified the controllers to reflect the changes

Source: https://medium.com/@kerimkkara/authentication-and-authorization-in-asp-net-core-a-comprehensive-guide-dfb8fb806ac7

### CODE REVIEW

File: CodebustersPAS-Server/Infrastructure/Migrations/20240924215557_refactor-database.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                defaultValue: "");
+
+            migrationBuilder.AddColumn<int>(
+                name: "StudentID",
+                table: "Users",
+                type: "INTEGER",
+                nullable: true);
+
+            migrationBuilder.AddColumn<string>(
+                name: "eamil",
```
Comment (Theos1010) [inline CodebustersPAS-Server/Infrastructure/Migrations/20240924215557_refactor-database.cs:39]: Is this a typo at line 39?

File: CodebustersPAS-Server/Infrastructure/Migrations/20240924215557_refactor-database.cs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            migrationBuilder.DropColumn(
+                name: "Password",
+                table: "Users");
+
+            migrationBuilder.DropColumn(
+                name: "StudentID",
+                table: "Users");
+
+            migrationBuilder.DropColumn(
+                name: "eamil",
```
Comment (Theos1010) [inline CodebustersPAS-Server/Infrastructure/Migrations/20240924215557_refactor-database.cs:62]: Typo at line 62?

File: CodebustersPAS-Server/Infrastructure/Migrations/DbContextModelSnapshot.cs [hunk 3]
```diff
@@ -70,11 +75,16 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
+                    b.Property<string>("Password")
+                        .IsRequired()
+                        .HasColumnType("TEXT");
+
                     b.Property<Guid?>("TeamId")
                         .HasColumnType("TEXT");
 
-                    b.Property<int>("type")
-                        .HasColumnType("INTEGER");
+                    b.Property<string>("eamil")
```
Comment (Theos1010) [inline CodebustersPAS-Server/Infrastructure/Migrations/DbContextModelSnapshot.cs]: Typo at line 85?

File: CodebustersPAS-Server/Infrastructure/Models/User.cs [hunk 4]
```diff
@@ -1,10 +1,10 @@
-﻿using Infrastructure.Enums;
-
+﻿
 namespace Infrastructure.Models {
     public class User {
         public Guid Id { get; set; }
-        public string FirstName { get; set; }
-        public string LastName { get; set; }
-        public UserType type { get; set; }
+        public required string eamil { get; set; }
```
Comment (Theos1010) [inline CodebustersPAS-Server/Infrastructure/Models/User.cs]: Typo at line 5?

File: CodebustersPAS-Server/Infrastructure/Migrations/20240924215557_refactor-database.cs (hunk 1 above)
Comment (LouisChoiniere) [reply CodebustersPAS-Server/Infrastructure/Migrations/20240924215557_refactor-database.cs:39]: yes lol I'll change it

**Review by tinaw1412 — APPROVED** [review]

**Review by OmarSelim01 — APPROVED** [review]

**Pull request merged**